packages/modal/src/Modal/Modal.tsx

```diff
@@ -111,51 +126,68 @@ const RootComponent = ({
         </VisuallyHidden.Root>
         <AnimatePresence>
           {open && (
-            <Overlay layer={layer}>
-              <Stack
-                as={motion.div}
-                initial={{
-                  top: `calc(var(--tgph-spacing-16) + var(--tgph-spacing-4) * ${layersLength - 1})`,
-                }}
-                animate={{
-                  top: isStacked
-                    ? `calc(var(--tgph-spacing-16) + var(--tgph-spacing-4) * ${layer} )`
-                    : "var(--tgph-spacing-16)",
-                }}
-                exit={{ top: 0 }}
-                transition={{ type: "spring", duration: 0.3, bounce: 0 }}
-                w="full"
-                justify="center"
-                style={{
-                  position: "fixed",
-                  left: 0,
-                  maxHeight: "calc(100vh - var(--tgph-spacing-32))",
-                  maxWidth: "calc(100vw - var(--tgph-spacing-8))",
-                  zIndex: `calc(var(--tgph-zIndex-modal) + ${layer})`,
-                }}
-                key={`container-${id}`}
-              >
-                <Stack
-                  direction="column"
-                  as={motion.div}
-                  animate={{
-                    scale: 1.02 - Math.abs(layersLength - layer) * 0.02,
-                    transformOrigin: "center center",
-                  }}
-                  transition={{ duration: 0.2, bounce: 0, type: "spring" }}
-                  maxW={props.maxW ?? "160"}
-                  w={props.w ?? "full"}
-                  bg="surface-1"
-                  border="px"
-                  rounded="4"
-                  shadow="3"
-                  key={`content-${id}`}
-                  {...props}
+            // We add the className "tgph" here so that styles within
+            // the portal get scoped properly to telegraph
+            <Portal.Root className="tgph">
```

Comment:
Added `Portal.Root` here. This makes it so every modal will exist in a portal and is much easier to manage for stacking as each modal is now at the same hierarchy in the DOM. I don't see any use case that we wouldn't want to use a portal here so I made it the default.